lxd/instance.go

```diff
@@ -174,6 +174,137 @@ func instanceCreateFromImage(s *state.State, r *http.Request, img *api.Image, ar
 	return inst, nil
 }
 
+func instanceRebuildFromImage(s *state.State, r *http.Request, inst instance.Instance, img *api.Image, op *operations.Operation) error {
+	// Validate the type of the image matches the type of the instance.
+	imgType, err := instancetype.New(img.Type)
+	if err != nil {
+		return err
+	}
+
+	if imgType != inst.Type() {
+		return fmt.Errorf("Requested image's type %q doesn't match instance type %q", imgType, inst.Type())
+	}
+
+	// Check if the image is available locally or it's on another member.
+	// Ensure we are the only ones operating on this image. Otherwise another instance created at the same
+	// time may also arrive at the conclusion that the image doesn't exist on this cluster member and then
+	// think it needs to download the image and store the record in the database as well, which will lead to
+	// duplicate record errors.
+	unlock := imageOperationLock(img.Fingerprint)
+
+	nodeAddress, err := s.DB.Cluster.LocateImage(img.Fingerprint)
```

Comment:
You'll see `node` mentioned in a few places when referring to cluster member.
However we never mention "node" in public facing docs/APIs, so for new code I think it would be good to get out of the habit of mentioning "node" too and instead use "member". Thanks